ssg/oval.py

```diff
@@ -98,7 +98,17 @@ def applicable_platforms(oval_file):
 
     platforms = []
     header = oval_generated_header("applicable_platforms", "5.11", "0.0.1")
-    body = process_file_with_macros(oval_file, {})
+    subst_dict = dict()
+
+    oval_filename_components = oval_file.split(os.path.sep)
+    if len(oval_filename_components) > 3:
+        subst_dict["rule_id"] = oval_filename_components[-3]
+    else:
+        msg = "Unable to get rule ID from OVAL path '{path}'".format(path=oval_file)
+        print(msg, file=sys.stderr)
+
+    body = process_file_with_macros(oval_file, subst_dict)
+
```

Comment:
What is this change good for?

Reply:
A test was failing due to `{{{ rule_id }}` usage in the OVAL snippet. Which is BTW strange, as that is not the only snippet that contains it.

Reply:
Usage of the `rule_id` substitution in the OVAL made a test that uses this functionality to fail. That change makes it passing again.